src/components/Picker/BasePicker.tsx

```diff
@@ -0,0 +1,278 @@
+import React, {useCallback, useContext, useEffect, useRef, useState} from 'react';
+import {HostComponent, MeasureLayoutOnSuccessCallback, StyleProp, View, ViewStyle} from 'react-native';
+import RNPickerSelect from 'react-native-picker-select';
+import _ from 'underscore';
+import usePrevious from '../../hooks/usePrevious';
+import styles from '../../styles/styles';
+import themeColors from '../../styles/themes/default';
+import FormHelpMessage from '../FormHelpMessage';
+import Icon from '../Icon';
+import * as Expensicons from '../Icon/Expensicons';
+import {ScrollContext} from '../ScrollViewWithContext';
+import Text from '../Text';
+
+type Props = {
+    /** BasePicker label */
+    label?: string;
+
+    /** Should the picker appear disabled? */
+    isDisabled: boolean;
+
+    /** Input value */
+    value?: string | number;
+
+    /** The items to display in the list of selections */
+    items: Array<{
+        value: string | number;
+        label: string;
+    }>;
+
+    /** Something to show as the placeholder before something is selected */
+    placeholder:
+        | {
+              value: string;
+              label: string;
+          }
+        | Record<string, never>;
+
+    /** Error text to display */
+    errorText: string;
+
+    /** Customize the BasePicker container */
+    containerStyles: StyleProp<ViewStyle>;
+
+    /** Customize the BasePicker background color */
+    backgroundColor?: string;
+
+    /** The ID used to uniquely identify the input in a Form */
+    inputID?: string;
+
+    /** Saves a draft of the input value when used in a form */
+    // eslint-disable-next-line react/no-unused-prop-types
+    shouldSaveDraft: boolean;
+
+    /** A callback method that is called when the value changes and it receives the selected value as an argument */
+    onInputChange: (value: string | number, index: number) => void;
+
+    /** Size of a picker component */
+    size: 'normal' | 'small';
+
+    /** An icon to display with the picker */
+    icon: (size: 'normal' | 'small') => JSX.Element;
+
+    /** Whether we should forward the focus/blur calls to the inner picker * */
+    shouldFocusPicker: boolean;
+
+    /** Callback called when click or tap out of BasePicker */
+    onBlur: () => void;
+
+    /** Additional events passed to the core BasePicker for specific platforms such as web */
+    additionalPickerEvents: (onMouseDown: any, onChange: any) => void;
+
+    /** Hint text that appears below the picker */
+    hintText: string;
+};
+
+const defaultProps: Omit<Props, 'items' | 'onInputChange'> = {
```

Comment:
@blazejkustra What do you think about this approach to defaultProps? The developer can either Omit or Pick which props from `Props` they are going to set on defaultProps.

Reply:
I have used a similar pattern to perform a kind of "prop inheritance" with the existing propTypes using `_.omit`. So I like this pattern, I think it's pretty easy and clear

Reply:
I like it 👌 @fabioh8010